test/integration/conftest.py

```diff
@@ -246,3 +267,16 @@ def upload_media(conservator, media):
         )
         media_ids.append(media_id)
     conservator.media.wait_for_processing(media_ids)
+
+
+def wait_for_dataset_commit(cns, dataset_id):
```

Comment:
Just an observation - it might be useful to have this function in the main codebase, possibly in `local_dataset`.

Reply:
Would it be worth adding a call to this method inside `Dataset.commit()`, just to avoid the possibility that user code triggers the race condition by creating a dataset then quickly deleting it again?

Reply:
I don't think so. A user can do stuff with datasets immediately without worrying about commits (e.g. adding frames, changing permissions, etc.) - I think a user creating a dataset and immediately deleting it is an edge case, and not worth delaying creation of a dataset over.

Reply:
Can you move this function to the main codebase, please?

Reply:
Moved into dataset.py.